jsk_pcl_ros_utils/src/pcd_capture_nodelet.cpp

```diff
@@ -0,0 +1,99 @@
+// -*- mode: c++; indent-tabs-mode: nil; -*-
+/*********************************************************************
+ * Software License Agreement (BSD License)
+ *
+ *  Copyright (c) 2016, JSK Lab
+ *  All rights reserved.
+ *
+ *  Redistribution and use in source and binary forms, with or without
+ *  modification, are permitted provided that the following conditions
+ *  are met:
+ *
+ *   * Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *   * Redistributions in binary form must reproduce the above
+ *     copyright notice, this list of conditions and the following
+ *     disclaimer in the documentation and/o2r other materials provided
+ *     with the distribution.
+ *   * Neither the name of the JSK Lab nor the names of its
+ *     contributors may be used to endorse or promote products derived
+ *     from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ *  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ *********************************************************************/
+
+#define BOOST_PARAMETER_MAX_ARITY 7
+
+#include "jsk_pcl_ros_utils/pcd_capture.h"
+#include "jsk_recognition_utils/pcl_conversion_util.h"
+#include <pcl/io/io.h>
+#include <pcl/point_types.h>
+#include <pcl/exceptions.h>
+#include <sensor_msgs/PointCloud2.h>
+
+namespace jsk_pcl_ros_utils
+{
+  void PCDCapture::onInit()
+  {
+    DiagnosticNodelet::onInit();
+    srv_ = boost::make_shared <dynamic_reconfigure::Server<Config> > (*pnh_);
+    dynamic_reconfigure::Server<Config>::CallbackType f = boost::bind (
+        &PCDCapture::configCallback, this, _1, _2);
+    srv_->setCallback (f);
+    sub_ = pnh_->subscribe("input", 1, &PCDCapture::callback, this);
+    onInitPostProcess();
+  }
+
+  void PCDCapture::subscribe()
+  {
+  }
```

Comment:
That is because this node will never be subscribed.
The parent class of DiagnosticNodelet is ConnectionBasedNodelet, and it subscribes only when being subscribed.
You don't need to use them in this case.